### PR TITLE
Mock array multiple metal layers pins

### PR DESCRIPTION
**Pull request by oharboe**
This test case is best articulated as a pull request. There are some issues to be worked out with pins on multiple metal layers.

### CODE REVIEW

**Comment by oharboe** [comment]
Broke out github issue for failure https://github.com/The-OpenROAD-Project/OpenROAD/issues/5853

**Comment by oharboe** [comment]
@maliberty PDN works now, but lots of DRC errors on the sides:

image


image

**Comment by maliberty** [comment]
Looks like you need https://github.com/The-OpenROAD-Project/OpenROAD/issues/5854 for the pins to be on track.

That said I wonder if we couldn't have routed these off-grid as there are no other wires to conflict with - @osamahammad21 thoughts?

image

**Comment by maliberty** [comment]
Nvm, that layer has LEF58_RIGHTWAYONGRIDONLY

**Comment by oharboe** [comment]
> Looks like you need [The-OpenROAD-Project/OpenROAD#5854](https://github.com/The-OpenROAD-Project/OpenROAD/issues/5854) for the pins to be on track.

Thanks! I'll keep this PR open and try again when it is merged.

**Comment by oharboe** [comment]
@maliberty FYI, works now.

**Comment by oharboe** [comment]
@maliberty I've reenabled pin placement annealing. That it is slow is a separate concern to this PR, issue filed.

File: flow/designs/asap7/mock-array/rules-base.json [hunk 1]
```diff
@@ -48,15 +48,15 @@
         "compare": "<="
     },
     "finish__timing__setup__ws": {
-        "value": 0.0,
+        "value": -15.92,
```
Comment (maliberty) [inline flow/designs/asap7/mock-array/rules-base.json]: Any insight into this degradation?

Comment (oharboe) [reply flow/designs/asap7/mock-array/rules-base.json]: I think we need the --annealing io fix first before we have something comparable, but that said things are quite different w.r.t. routing and metal layers. 15ns isnt much, CTS has more variations, etc

Comment (maliberty) [reply flow/designs/asap7/mock-array/rules-base.json]: I hope its ps

**Review by maliberty — APPROVED** [review]

**Comment by oharboe** [comment]
@maliberty Annealing runs "forever" (usually seconds, now hours). Seperate issue filed.

**Comment by oharboe** [comment]
@maliberty times out after 6 hours in PPL. Will retry after https://github.com/The-OpenROAD-Project/OpenROAD/issues/5847 is fixed.

**Comment by oharboe** [comment]
@maliberty @eder-matheus Lots of new DRC errors. It might be worth looking at [recent pin placement changes](https://github.com/The-OpenROAD-Project/OpenROAD/issues/5847).

If the mirror option was broken, it could present itself as these DRC errors:

image


image

**Comment by oharboe** [comment]
@maliberty @eder-matheus These pins should be connected to the bottom of the element, so the problem is in the pin placement of the element, it would seem.

image

**Comment by oharboe** [comment]
@maliberty @eder-matheus `io_outs_down[9]` should be at the bottom.

image

to reproduce with latest OpenROAD master:

[io_placement.tar.gz](https://github.com/user-attachments/files/17349220/io_placement.tar.gz)

**Comment by maliberty** [comment]
It is easier to track issues when filed as such.  @eder-matheus please take a look.

**Comment by oharboe** [comment]
> It is easier to track issues when filed as such. @eder-matheus please take a look.

Filed here: https://github.com/The-OpenROAD-Project/OpenROAD/issues/5926

**Comment by oharboe** [comment]
@maliberty @eder-matheus Works :-) Ready to merge now, saw pr-head passed for mock-array by drilling into the run.

**Comment by oharboe** [comment]
@maliberty There are unrelated failures in pr-head. Those are presumably there because of my OpenROAD update. I will merge latest master with this PR after OpenROAD has been updated in master.

**Comment by maliberty** [comment]
>  I will merge latest master with this PR after OpenROAD has been updated in master.

Is there another PR doing this?

**Comment by oharboe** [comment]
> > I will merge latest master with this PR after OpenROAD has been updated in master.
> 
> Is there another PR doing this?

The bot that updates OpenROAD in ORFS?

**Comment by maliberty** [comment]
Please address the conflicts

**Comment by oharboe** [comment]
> Please address the conflicts

Fixed.

**Pull request merged**